db/migrate/20210407104300_downcase_mac_addresses.rb

```diff
@@ -0,0 +1,8 @@
+class DowncaseMacAddresses < ActiveRecord::Migration[6.0]
+  def up
+    execute 'UPDATE nics SET mac=LOWER(mac)'
```

Comment:
I'd like to as a more experienced Rails dev to comment on the right way to do this. I'd say that using the ORM is normally a good thing but maybe migrations are different.

Reply:
Using the ORM in migration can be dangerous: if for any reason the model vanishes, the migration is broken.

Reply:
I am far from an experience RoR dev, but generally I like this because we only support PostgreSQL and this will be much faster than via ORM. Our biggest customers have up to 100,000 nodes and that is not many records to update, but some customers do have many NICs in their database created by accident (unfiltered container hosts) and then amount of records could be actually higher.

Reply:
Usually using the ORM class is better, then direct SQL query, because individual setups can choose to use prefixed table names, but that is not an issue for us, so this is ok to use.

the best for this is usually:
```
Nic::Base.update_all(mac: Nic::Base.arel_table[:mac].lower)
```
There is quite some magic, that rails does and can be useful, but none of that is used here I'd bet.

The case where class dissapear is valid, but in that case you want the migration to fail, so you know what migrations were touching that class table so you can deal with them :))

Reply:
I'll revert and update the tests today.